assets/js/modules/analytics-4/utils/withCustomDimensions.js

```diff
@@ -270,29 +308,31 @@ export default function withCustomDimensions( options = {} ) {
 			useEffect( () => {
 				if (
 					! customDimensions ||
-					! isInvalidCustomDimensionError( reportError ) ||
+					! hasInvalidCustomDimensionError ||
 					isSyncingAvailableCustomDimensions
 				) {
 					return;
 				}
 
-				// Clear report error so that the fetch sync action isn't
-				// triggered multiple times.
-				clearError( 'getReport', [ reportOptions ] );
+				scheduleSyncAvailableCustomDimensions();
 
-				// Sync available custom dimensions.
-				fetchSyncAvailableCustomDimensions().then( () => {
-					// Invalidate report request so that it is re-fetched.
-					invalidateResolution( 'getReport', [ reportOptions ] );
+				// Clear report errors so that the useEffect isn't
+				// triggered multiple times.
+				Promise.all(
+					invalidCustomDimensionReportOptions.map( ( args ) => {
+						return clearError( 'getReport', [ args ] );
+					} )
+				).finally( () => {
+					// Sync available custom dimensions.
+					scheduleSyncAvailableCustomDimensions();
```

Comment:
Since the above `Promise.all` is only clearing errors and can't throw, let's update this to use `await` and then call this afterwards without using `finally` or `then`.

Reply:
Updated, thanks! I had only added it to avoid using `async-await` inside a `useEffect` 😅